js/foundation.offcanvas.js

```diff
@@ -87,8 +89,14 @@ class OffCanvas {
       'keydown.zf.offcanvas': this._handleKeyboard.bind(this)
     });
 
-    if (this.options.closeOnClick && this.$exiter.length) {
-      this.$exiter.on({'click.zf.offcanvas': this.close.bind(this)});
+    // If we have an overlay and close on click, let it close the off canvas menu.
+    if (this.options.closeOnClick && this.options.contentOverlay && this.$overlay.length) {
+      this.$overlay.on({'click.zf.offcanvas': this.close.bind(this)});
+    }
+
+    // If content overlay is false but close on click is true, close via click on content.
+    if (this.options.closeOnClick && this.options.contentOverlay !== true) {
+      $('[data-off-canvas-content]').on({'click.zf.offcanvas': this.close.bind(this)});
```

Comment:
Conceptually, we're walking down a path of 
1. Do we closeOnClick
2. What do we apply the handler to?

So wouldn't the code be clearer if we combine these two if statements into something like:

```
if (this.options.closeOnClick) {                                            
    var $target = this.options.contentOverlay ? this.$overlay : $('[data-off-canvas-content]');
    $target.on({'click.zf.offcanvas': this.close.bind(this)});                
}
```